pkg/osd_cluster_upgrader/upgrader.go

```diff
@@ -167,8 +177,37 @@ func EnsureExtraUpgradeWorkers(c client.Client, cfg *osdUpgradeConfig, s scaler.
 	return isScaled, nil
 }
 
+// ExternalDependencyAvailabilityCheck validates that external dependencies of the upgrade are available.
+func ExternalDependencyAvailabilityCheck(c client.Client, cfg *osdUpgradeConfig, s scaler.Scaler, dsb drain.NodeDrainStrategyBuilder, metricsClient metrics.Metrics, m maintenance.Maintenance, cvClient cv.ClusterVersion, nc eventmanager.EventManager, upgradeConfig *upgradev1alpha1.UpgradeConfig, machinery machinery.Machinery, availabilityCheckers []ac.AvailabilityChecker, logger logr.Logger) (bool, error) {
+	upgradeCommenced, err := cvClient.HasUpgradeCommenced(upgradeConfig)
+	if err != nil {
+		return false, err
+	}
+	desired := upgradeConfig.Spec.Desired
+	if upgradeCommenced {
+		logger.Info(fmt.Sprintf("ClusterVersion is already set to Channel %s Version %s, skipping %s", desired.Channel, desired.Version, upgradev1alpha1.UpgradeScaleUpExtraNodes))
+		return true, nil
+	}
+
+	if len(availabilityCheckers) == 0 {
+		logger.Info("No external dependencies configured for availability checks. Skipping.")
+		return true, nil
+	}
+
+	for _, check := range availabilityCheckers {
+		logger.Info(fmt.Sprintf("Checking availability check for %T", check))
+		err := check.AvailabilityCheck()
+		if err != nil {
+			logger.Info(fmt.Sprintf("Failed availability check for %T", check))
+			return false, err
+		}
+		logger.Info(fmt.Sprintf("Availability check complete for %T", check))
+	}
```

Comment:
Substituting `%v` for `%T` here gives more useful and actionable info.

eg. it results in
```
Checking availability check for &{[https://quay.io/health https://apiasdfasdf.stage.openshift.com] 10s}
```

instead of:
```
Checking availability check for *availabilitychecks.HTTPAvailabilityChecker
```

Reply:
I wanted the implementation to be raised in the logs for future cases where multiple implementations may exist. The failed url/s are returned to the caller via `err`